### PR TITLE
Remove colour specification from some examples

### PR DESCRIPTION
**Pull request by dstansby**
A subset of #13369, only the ones where removing the colours altogether is fine in my opinion.

### CODE REVIEW

File: examples/statistics/barchart_demo.py [hunk 1]
```diff
@@ -28,9 +28,9 @@
 
 fig, ax = plt.subplots()
 rects1 = ax.bar(ind - width/2, men_means, width, yerr=men_std,
-                color='SkyBlue', label='Men')
+                label='Men')
```
Comment (jklymak) [inline examples/statistics/barchart_demo.py]: Maybe sexist, but one could argue that the colors had gender associations that help in reading the chart....

Comment (dstansby) [reply examples/statistics/barchart_demo.py]: The default colour cycle makes them different colours anyway: https://1164-74355158-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/statistics/barchart_demo.html#sphx-glr-gallery-statistics-barchart-demo-py

Comment (jklymak) [reply examples/statistics/barchart_demo.py]: Sure, but blue/light red have are often deemed masculine/feminine.  OTOH, I don't really feel strongly we need to perpetuate that.

Comment (dstansby) [reply examples/statistics/barchart_demo.py]: I agree, which is why I removed the explicit colours.

**Pull request merged**